frontends/web/src/routes/exchange/bitrefill.tsx

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright 2025 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Header } from '@/components/layout';
+import { Spinner } from '@/components/spinner/Spinner';
+import { AccountCode, IAccount, proposeTx, sendTx, TTxInput } from '@/api/account';
+import { findAccount, isBitcoinOnly } from '@/routes/account/utils';
+import { useDarkmode } from '@/hooks/darkmode';
+import { getConfig } from '@/utils/config';
+import style from './iframe.module.css';
+import { i18n } from '@/i18n/i18n';
+import { alertUser } from '@/components/alert/Alert';
+import { parseExternalBtcAmount } from '@/api/coins';
+import { useLoad } from '@/hooks/api';
+import { BitrefillTerms } from '@/components/terms/bitrefill-terms';
+import { getBitrefillInfo } from '@/api/exchanges';
+
+// Map languages supported by Bitrefill
+export const localeMapping: Readonly<Record<string, string>> = {
+  en: 'en',
+  de: 'de',
+  fr: 'fr',
+  es: 'es',
+  it: 'it',
+  pt: 'pt',
+  ja: 'ja',
+  zh: 'zh-Hans'
+};
+
+// Map coins supported by Bitrefill
+const coinMapping: Readonly<Record<string, string>> = {
+  btc: 'bitcoin',
+  tbtc: 'bitcoin',
+  ltc: 'litecoin',
+  eth: 'ethereum',
+  usdt: 'usdt_erc20',
+  usdc: 'usdc_erc20',
+};
+
+type TProps = {
+  accounts: IAccount[];
+  code: AccountCode;
+};
+
+const getURLOrigin = (uri: string): string | null => {
+  try {
+    return new URL(uri).origin;
+  } catch (e) {
+    return null;
+  }
+};
```

Comment:
nit: we have same (similar?) function in btcdirect, this could be shared in a new  `utils/url.ts or similar.